src/main/scala/viper/silver/cfg/silver/SilverCfg.scala

```diff
@@ -82,6 +82,65 @@ class SilverCfg(val blocks: Seq[SilverBlock], val edges: Seq[SilverEdge], val en
                     entry: SilverBlock = entry,
                     exit: Option[SilverBlock] = exit): SilverCfg =
     SilverCfg(blocks, edges, entry, exit)
+
+  private def findJoinPoint(queueInit: Iterable[SilverBlock],
+                            visitedInit: Iterable[SilverBlock],
+                            // We never enqueue loopHeads which we already have seen.
+                            // This would lead to non-termination.
+                            loopHeadsSeen: Iterable[SilverBlock],
+                            getNext: SilverBlock => Iterable[SilverBlock])
+  : (Option[SilverBlock], mutable.Map[SilverBlock, SilverBlock]) = {
+
+    var queue = mutable.Queue.from(queueInit)
+    var visited: Vector[SilverBlock] = Vector.from(visitedInit)
+    val map = mutable.Map[SilverBlock, SilverBlock]()
+    var loopHeads: Vector[SilverBlock] = Vector.from(loopHeadsSeen)
+
+    while (queue.nonEmpty) {
+      val curr = queue.dequeue()
+      val visitNext = curr match {
+        case curr =>
```

Comment:
1. Is this the only case? Why match at all?
2. `curr` from line 102 shadows `curr` from line 100, which is not recommended. If you want to match against the `curr` from line 100, the syntax would be  "case \`curr\` => ...", if I'm not mistaken.